src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1

```diff
@@ -238,6 +238,18 @@ Rule 'Azure.AKS.UptimeSLA' -Ref 'AZR-000285' -Type 'Microsoft.ContainerService/m
     $Assert.Contains($TargetObject, 'sku.tier', 'Paid');
 }
 
+# Synopsis: AKS clusters should use ephemeral OS disks which can provide lower read/write latency, along with faster node scaling and cluster upgrades.
+Rule 'Azure.AKS.EphemeralOSDisk' -Ref 'AZR-000287' -Level Warning -Type 'Microsoft.ContainerService/managedClusters', 'Microsoft.ContainerService/managedClusters/agentPools' -Tag @{ release = 'GA'; ruleSet = '2022_09' } {
+    $agentPools = @(GetAgentPoolProfiles);
+    if ($agentPools.Length -eq 0) {
+        return $Assert.Pass();
+    }
+    foreach ($agentPool in $agentPools) {
+        $Assert.In($agentPool, 'osDiskType', 'Ephemeral').
+        ReasonIf($agentPool.osDiskType, $LocalizedData.AKSEphemeralOSDiskNotConfigured);
```

Comment:
@bengeset96 We may cause some necessary false positives as you've noted in the docs. Thinking more on this rule, I think it matter less that `Ephemeral` is set, and more that `Managed` has not be explicitly set. Since if a preference is not set either way then ephemeral disks will be used if possible. What do you think?

Another way to address the rule would be to build a mapping of ephemeral disk sizes, but I don't think this would be every effective to maintain longer term.

```suggestion
        $Assert.HasDefaultValue($agentPool, 'osDiskType', 'Ephemeral').
        ReasonIf($agentPool.osDiskType, $LocalizedData.AKSEphemeralOSDiskNotConfigured);
```

Reply:
I fully agree. Actually I should have taught that way around it in the first place, this is much better indeed.

Orginially my idea was the build a mapping, but after some tinkering I choose not to go this path. Reason for this was simply due to that AKS teams in organizations often tend to cattle alot and rebuild from scratch time to time, tweak their configuration, and it would have been unfortunate if the mapping already was outdated and did not give accurate results. I just think for now it will lead to management overhead for us and something that would have to be maintained regularly. I feel like `HasDefaultValue` will be a good trade-off.